### PR TITLE
[NUI] Apply Tizen 7.0 UX on Pagination

### PR DESCRIPTION
**Pull request by Seoyeon2Kim**
Signed-off-by: Seoyeon Kim <seoyeon2.kim@samsung.com>

### Description of Change ###
- The latest Pagination UX has been applied. (Based on API10)
 The size and the default images are changed.

- The type of Pagination indicator is changed from `ImageVisual`
 to `ImageView`.
 Now, the indicators get highlight on AT-SPI.

### API Changes ###
- N/A

### CODE REVIEW

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 1]
```diff
@@ -512,18 +507,26 @@ public override void OnInitialize()
...
             {
                 Name = "Container",
                 ParentOrigin = Tizen.NUI.ParentOrigin.CenterLeft,
                 PivotPoint = Tizen.NUI.PivotPoint.CenterLeft,
                 PositionUsesPivotPoint = true,
+                Layout = new LinearLayout()
+                {
+                    LinearOrientation = LinearLayout.Orientation.Horizontal,
+#pragma warning disable CA2000
+                    CellPadding = new Size(indicatorSpacing, 0),
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Controls/Pagination.cs:520]: The type of CellPadding is Size2D.
Is this related to CA2000?

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 2]
```diff
@@ -512,18 +507,26 @@ public override void OnInitialize()
...
+#pragma warning restore CA2000
+                },
+                Padding = new Extents(8, 8, 0, 0),
             };
             this.Add(container);
 
             //TODO: Apply color properties from PaginationStyle class.
-            indicatorColor = new Color(1.0f, 1.0f, 1.0f, 0.5f);
-            selectedIndicatorColor = new Color(1.0f, 1.0f, 1.0f, 1.0f);
+            indicatorColor = new Color("#FAFAFA");
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Controls/Pagination.cs:528]: It would be better if these values are defined in theme later.

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 3]
```diff
@@ -654,17 +705,23 @@ private void UpdateVisual()
                 return;
             }
 
+            if (container != null && container.Layout is LinearLayout)
+            {
+#pragma warning disable CA2000
+                (container.Layout as LinearLayout).CellPadding = new Size(indicatorSpacing, 0);
```
Comment (jaehyun0cho) [inline src/Tizen.NUI.Components/Controls/Pagination.cs:711]: The type of CellPadding is Size2D.

**Review by jaehyun0cho — COMMENTED** [review]
It would be better if subcomponents' sizes and color values are defined in theme later.

**Comment by dongsug-song** [comment]
could you resolve the warning below?

src/Tizen.NUI.Components/Controls/Pagination.cs#L53
Use nameof in place of string literal 'IndicatorImageUrl'

**Comment by bshsqa** [comment]
Please remove some magic numbers.

File: src/Tizen.NUI.Components/Controls/Pagination.cs (hunk 1 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:520]: Actually, no. It's not related to CA2000. The warning message is to call Dispose() or use `using` statement instead.
But, I'll remove warning suppression once.

File: src/Tizen.NUI.Components/Controls/Pagination.cs (hunk 3 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:711]: Done.

**Comment by Seoyeon2Kim** [comment]
> could you resolve the warning below?
> 
> src/Tizen.NUI.Components/Controls/Pagination.cs#L53 Use nameof in place of string literal 'IndicatorImageUrl'

Got it!

File: src/Tizen.NUI.Components/Controls/Pagination.cs (hunk 1 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs]: Oh, the error message is because of the type mismatching! 
Thank you for checking it!! :+1:

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 4]
```diff
@@ -74,11 +74,11 @@ public partial class Pagination : Control, IAtspiValue
             return ((Pagination)bindable).indicatorSpacing;
         });
 
-        private VisualView container;
-        private Size indicatorSize = new Size(10, 10);
+        private View container;
+        private Size indicatorSize = new Size(64, 8);
```
[the remark this answers is not in the document]
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:78]: Done

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 5]
```diff
@@ -74,11 +74,11 @@ public partial class Pagination : Control, IAtspiValue
...
         private Selector<string> indicatorImageUrl;
-        private int indicatorSpacing;
-        private List<ImageVisual> indicatorList = new List<ImageVisual>();
+        private int indicatorSpacing = 16;
```
[the remark this answers is not in the document]
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:80]: Done

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 6]
```diff
@@ -324,9 +323,7 @@ private Color InternalIndicatorColor
                         continue;
                     }
 
-                    ImageVisual indicator = indicatorList[i];
-                    indicator.MixColor = indicatorColor;
-                    indicator.Opacity = indicatorColor.A;
+                    indicatorList[i].Color = indicatorColor;
```
[the remark this answers is not in the document]
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:326]: Changed to use Color of View, which is Color property of DALi Actor.

File: src/Tizen.NUI.Components/Controls/Pagination.cs [hunk 7]
```diff
@@ -512,18 +507,26 @@ public override void OnInitialize()
...
+                },
+                Padding = new Extents(8, 8, 0, 0),
             };
             this.Add(container);
 
             //TODO: Apply color properties from PaginationStyle class.
-            indicatorColor = new Color(1.0f, 1.0f, 1.0f, 0.5f);
-            selectedIndicatorColor = new Color(1.0f, 1.0f, 1.0f, 1.0f);
+            indicatorColor = new Color("#FAFAFA");
+            selectedIndicatorColor = new Color("#FFA166");
```
[the remark this answers is not in the document]
Comment (Seoyeon2Kim) [reply src/Tizen.NUI.Components/Controls/Pagination.cs:527]: Done

**Review by jaehyun0cho — APPROVED** [review]
LGTM
